pyproject.toml

```diff
@@ -71,7 +71,8 @@ ocsp = [
     "service_identity>=18.1.0",
 ]
 snappy = [
-    "python-snappy",
+    "python-snappy;implementation_name!='pypy' or implementation_version>'7.3.11'",
+    "python-snappy<0.7.0;implementation_name=='pypy' and implementation_version<='7.3.11'"
```

Comment:
We should make this change only in the test suite. That's what we usually do for issues like this. That way if python-snappy fixes this bug then we won't need to update pymongo.

Reply:
Updated